aten/src/ATen/native/Normalization.cpp

```diff
@@ -202,14 +202,18 @@ std::tuple<Tensor,Tensor> batch_norm_cpu_update_stats_template(
     double momentum, double eps, Tensor& save_mean, Tensor& save_var_transform) {
 
   using accscalar_t = at::acc_type<scalar_t, false>;
+  using opmath_t = std::conditional_t<std::is_same_v<param_t, at::Half>, at::opmath_type<param_t>, param_t>;
 
   int64_t n_input = input.size(1);
   TORCH_CHECK(input.numel() != 0, "input tensor must have at least one element, but got input_sizes = ", input.sizes());
   int64_t n = input.numel() / n_input;
 
   bool all_contiguous = is_contiguous(input);
   constexpr bool mixed_type = !std::is_same<scalar_t, param_t>::value;
-  const auto dtype = mixed_type ? kFloat : input.scalar_type();
+  auto dtype = mixed_type ? kFloat : input.scalar_type();
+  if (dtype == kHalf) {
+    dtype = kFloat;
+  }
```

Comment:
Add some comments here to explain why we are doing so?

Reply:
Added comments.